### PR TITLE
[Search] SearchProvider and Tree Search enhancements/fixes

### LINKED ISSUE #3399: [Search] Search will fail to index items using the proper provider
Need to switch from modelService to objectService in GenericSearchProvider to handle cases where the wrong provider was being used.

# Testing
- include My Items plugin
- use CouchDB persistence
- Modify the My Items folder name and save (should then save to couch)
- Refresh and observe that the My Items folder (with the changed name) is being indexed for search

### PR DESCRIPTION
**Pull request by jvigliotta**
Added the objectService to the GenericSearchProvider to handle cases where the wrong provider was being used to get the object/model. Also small refactor to the navigation tree to handle search more appropriately (searching indicator, clearing children, etc).

closes #3399 

## Author Checklist
Changes address original issue? Yes
Unit tests included and/or updated with changes? N/A Legacy
Command line build passes? Yes
Changes have been smoke-tested? Yes
Testing instructions included? Yes

### CODE REVIEW

File: platform/search/bundle.js [hunk 1]
```diff
@@ -109,7 +109,8 @@ define([
                             "topic",
                             "GENERIC_SEARCH_ROOTS",
                             "USE_LEGACY_INDEXER",
-                            "openmct"
+                            "openmct",
+                            "objectService"
```
Comment (deeptailor) [inline platform/search/bundle.js:113]: lets add this in place of modelService, since we are replacing it

**Review by deeptailor — CHANGES_REQUESTED** [review]
just one callout so far

File: platform/search/bundle.js (hunk 1 above)
Comment (jvigliotta) [reply platform/search/bundle.js]: updated!

**Review by deeptailor — APPROVED** [review]
LGTM

**Comment by deeptailor** [comment]
@jvigliotta - were there any legacy tests that need updating? We should not be adding tests to legacy code but we should update them if needed

**Comment by deeptailor** [comment]
## Reviewer Checklist
Changes appear to address issue? Yes
Appropriate unit tests included? N/A
Code style and in-line documentation are appropriate? Yes
Commit messages meet standards? Yes
Has associated issue been labelled unverified? (only applicable if this PR closes the issue)Yes

**Pull request merged**